### DISCUSSION TITLE
How to add trame toolbar?

### DISCUSSION THREAD

**Discussion by LTMeyer** [edsaac/stpyvista#31] [Q&A] [answered]
Hello, thank you for the great work.

I understand `stpyvista` supports `trame` backend. How could I enable the `trame` toolbar there is with pyvista visualization in jupyter notebooks? I'm referring to the toolbar we see in the example below
![image](https://github.com/user-attachments/assets/529f7584-0aea-4146-b519-4ddb066ed4db)

**Comment by edsaac** [comment, marked as answer]
`stpyvista` uses `trame` (or `panel`) to export the 3D view to a standalone html that is embedded in Streamlit.  All the nice tidbits on top that you see in Jupyter come from a full trame application that `pyvista` serves on the backend. However, serving such application in parallel and connect to it with Streamlit is not possible (see this closed PR: https://github.com/streamlit/streamlit/pull/4677). 

If your application relies heavily on 3D visualizations and interaction with those , I recommend to move your application to [`trame`](https://kitware.github.io/trame/).

**Reply by LTMeyer** [reply]
Thank you for the quick and clear reply. As I am developing a prototype for Hugging Face hub, which only supports gradio and streamlit and not trame (c.f. [HF Spaces](https://huggingface.co/docs/hub/spaces-overview#creating-a-new-space)), I will stick to `stpyvista` and the current working. 

Thank you.
